src/service_daemon.rs

```diff
@@ -740,76 +782,77 @@ impl Zeroconf {
+                service_info.remove_ipaddr(addr);
             }
         }
     }
 
     /// Check for IP changes and update intf_socks as needed.
     fn check_ip_changes(&mut self) {
-        // Get the current IPv4 interfaces.
-        let my_ifv4addrs = my_ipv4_interfaces();
+        // Get the current interfaces.
+        let my_ifaddrs = my_ip_interfaces();
 
         // Remove unused sockets in the poller.
         let deleted_addrs = self
             .intf_socks
             .iter()
             .filter_map(|(_, if_sock)| {
-                if !my_ifv4addrs.contains(&if_sock.intf) {
+                if !my_ifaddrs.contains(&if_sock.intf) {
                     if let Err(e) = self.poller.delete(&if_sock.sock) {
                         error!("check_ip_changes: poller.delete {:?}: {}", &if_sock.intf, e);
                     }
-                    Some(if_sock.intf.ip)
+                    Some(if_sock.intf.ip())
                 } else {
                     None
                 }
             })
-            .collect::<Vec<Ipv4Addr>>();
+            .collect::<Vec<IpAddr>>();
 
         // Remove deleted addrs from my services that enabled `addr_auto`.
-        for ipv4 in deleted_addrs.iter() {
-            self.del_addr_in_my_services(ipv4);
-            self.notify_monitors(DaemonEvent::Ipv4Del(*ipv4));
+        for ip in deleted_addrs.iter() {
+            self.del_addr_in_my_services(ip);
+            self.notify_monitors(DaemonEvent::IpDel(*ip));
         }
 
         // Keep the interfaces only if they still exist.
-        self.intf_socks
-            .retain(|_, v| my_ifv4addrs.contains(&v.intf));
+        self.intf_socks.retain(|_, v| my_ifaddrs.contains(&v.intf));
 
         // Add newly found interfaces.
-        for intf in my_ifv4addrs {
+        for intf in my_ifaddrs {
             // Skip existing interfaces.
-            if self.intf_socks.get(&intf.ip).is_some() {
+            if self.intf_socks.get(&intf.ip()).is_some() {
                 continue;
             }
 
             // Bind the new interface.
-            let new_ip = intf.ip;
-            let sock = match new_socket_bind(&new_ip) {
+            let new_ip = intf.ip();
+            let sock = match new_socket_bind(&intf) {
                 Ok(s) => {
-                    debug!("check_ip_changes: bind {}", &intf.ip);
+                    debug!("check_ip_changes: bind {}", &intf.ip());
                     s
                 }
                 Err(e) => {
-                    debug!("bind a socket to {}: {}. Skipped.", &intf.ip, e);
+                    debug!("bind a socket to {}: {}. Skipped.", &intf.ip(), e);
                     continue;
                 }
             };
 
             // Add the new interface into the poller.
-            let key = u32::from(new_ip) as usize;
+            let key = self.id_count;
+            self.id_count += 1;
+            let _ = self.poll_ids.insert(key, new_ip);
```

Comment:
If you combine them into a method, then don't need to repeat the steps here, and just make a call, something like `let key = self.add_poll(new_ip)`.